src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -297,19 +301,19 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 			}
 		}
 
-		const absoluteCursorY = buffer.baseY + buffer.cursorY;
 		let value = commandLine;
 		let ghostTextIndex = -1;
-		let cursorIndex: number;
-		if (absoluteCursorY === commandStartY) {
-			cursorIndex = this._getRelativeCursorIndex(this._commandStartX, buffer, line);
-		} else {
-			cursorIndex = commandLine.trimEnd().length;
+		if (cursorIndex === undefined) {
+			if (absoluteCursorY === commandStartY) {
+				cursorIndex = this._getRelativeCursorIndex(this._commandStartX, buffer, line);
+			} else {
+				cursorIndex = commandLine.trimEnd().length;
+			}
 		}
 
 		// Detect ghost text by looking for italic or dim text in or after the cursor and
 		// non-italic/dim text in the cell closest non-whitespace cell before the cursor
-		if (absoluteCursorY === commandStartY && buffer.cursorX > 1) {
+		if (this._shellType !== GeneralShellType.PowerShell || absoluteCursorY === commandStartY && buffer.cursorX > 1) {
 			// Ghost text in pwsh only appears to happen on the cursor line
```

Comment:
Probably my comment, but it's a little hard to understand. I might have meant command start line? That's uncorrect though given the above example😅